### PR TITLE
Fix AD0001: Named Attribute Arguments in S6930

### LINKED ISSUE #9193: Fix AD0001: Named Attribute Arguments in S6930
### Description
https://rules.sonarsource.com/csharp/RSPEC-6930/
CSC : warning AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.BackslashShouldBeAvoidedInAspNetRoutes' threw an exception of type 'System.InvalidOperationException' with message 'Sequence contains no elements'.

### Repro steps

I don't have, because can't find backslash. but it happens on project which don't have startup file and controllers

### Related information

* SonarLint 7.8.0.88494
* Visual Studio 2022 17.9.6
* .net 8.0.204
* SonarAnalyzer.CSharp 9.24.0.89429
* Windows

### PR DESCRIPTION
**Pull request by zsolt-kolbay-sonarsource**
Fixes #9193

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/Rules/AspNet/BackslashShouldBeAvoidedInAspNetRoutesBase.cs [hunk 1]
```diff
@@ -42,7 +44,8 @@ public abstract class BackslashShouldBeAvoidedInAspNetRoutesBase<TSyntaxKind> :
 
     protected void Check(SonarSyntaxNodeReportingContext c)
     {
-        if (Language.Syntax.NodeExpression(c.Node) is { } expression
+        if (!IsNamedAttributeArgument(c.Node)
+            && Language.Syntax.NodeExpression(c.Node) is { } expression
             && Language.FindConstantValue(c.SemanticModel, expression) is string constantRouteTemplate
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/Rules/AspNet/BackslashShouldBeAvoidedInAspNetRoutesBase.cs:49]: We aslodo have the `TryGetAttributeValue` extension method for such extractions.

File: analyzers/tests/SonarAnalyzer.Test/TestCases/AspNet/BackslashShouldBeAvoidedInAspNetRoutes.AspNet4x.cs [hunk 2]
```diff
@@ -110,3 +110,19 @@ void Test()
...
+// https://github.com/SonarSource/sonar-dotnet/issues/9193
+namespace AttributeWithNamedArgument
+{
+    [AttributeUsage(AttributeTargets.All)]
+    public class MyAttribute : Attribute
+    {
+        public string Name { get; set; }
+    }
+
+    public class MyController: Controller
```
Comment (antonioaversa) [inline analyzers/tests/SonarAnalyzer.Test/TestCases/AspNet/BackslashShouldBeAvoidedInAspNetRoutes.AspNet4x.cs:123]: `MyController : Controller` (space before `:`).

File: analyzers/src/SonarAnalyzer.Common/Rules/AspNet/BackslashShouldBeAvoidedInAspNetRoutesBase.cs [hunk 3]
```diff
@@ -42,7 +44,8 @@ public abstract class BackslashShouldBeAvoidedInAspNetRoutesBase<TSyntaxKind> :
 
     protected void Check(SonarSyntaxNodeReportingContext c)
     {
-        if (Language.Syntax.NodeExpression(c.Node) is { } expression
+        if (!IsNamedAttributeArgument(c.Node)
```
Comment (antonioaversa) [inline analyzers/src/SonarAnalyzer.Common/Rules/AspNet/BackslashShouldBeAvoidedInAspNetRoutesBase.cs:47]: Isn't this going to break the following test case?
```cs
public class WithRouteOnNamedParameter : Controller
{
    [Route(template: "A\\[action]\\B")] // Noncompliant
    //               ^^^^^^^^^^^^^^^^
    public IActionResult Index() => View();
}
```

**Review by antonioaversa — APPROVED** [review]
LGTM
Please have a look at the following two comments left:
- minor, non blocking: here
- more important: [here](https://github.com/SonarSource/sonar-dotnet/pull/9199/files#r1577942536)

**Pull request merged**